packages/core/src/hub.ts

```diff
@@ -480,7 +480,11 @@ export class Hub implements HubInterface {
     if (sentry && sentry.extensions && typeof sentry.extensions[method] === 'function') {
       return sentry.extensions[method].apply(this, args);
     }
-    __DEBUG_BUILD__ && logger.warn(`Extension method ${method} couldn't be found, doing nothing.`);
+    throw new Error(`Tracing extension '${method}' has not been added. Call 'addTracingExtensions' before calling 'init':
```

Comment:
Could we only throw this error in debug mode? Also this applies to all extensions, perhaps we move this into `hub.startTransaction`.